File: problems/1128/solution.py
from typing import List

class Solution:
    def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:
        myDict = {}
        result = 0
        for d in dominoes:
            if d[0] < d[1]:
                domino = str(d[0]) + str(d[1])
            else:
                domino = str(d[1]) +  str(d[0])
            if domino not in myDict:
                myDict[domino] = 1
            else:
                result += myDict[domino]
                myDict[domino] += 1
        return result
    
if __name__ == "__main__":
    dominoes = [[1,2],[1,2],[1,1],[1,2],[2,2]]
    print(Solution().numEquivDominoPairs(dominoes=dominoes))

File: problems/1128/solution.cpp
#include <iostream>
#include <vector>
#include <string>
#include <unordered_map>
using namespace std;

class Solution
{
public:
    int numEquivDominoPairs(vector<vector<int>> &dominoes)
    {
        unordered_map<int, int> myMap;
        int domino;
        int result = 0;
        for (const auto &d : dominoes)
        {
            if (d[0] > d[1])
            {
                domino = d[1] * 10 + d[0];
            }
            else
            {
                domino = d[0] * 10 + d[1];
            }
            if (myMap.find(domino) != myMap.end())
            {
                result += myMap[domino];
                myMap[domino] += 1;
            }
            else
            {
                myMap[domino] = 1;
            }
        }
        return result;
    }
};

int main()
{
    Solution sol;
    vector<vector<int>> dominos = {{1, 2}, {2, 1}, {3, 4}, {5, 6}};
    cout << sol.numEquivDominoPairs(dominos);
}